core/services/workflows/metering/metering.go

```diff
@@ -261,16 +253,21 @@ func ByResource(
 	amount decimal.Decimal,
 ) func(string, *Report) ([]capabilities.SpendLimit, error) {
 	return func(ref string, r *Report) ([]capabilities.SpendLimit, error) {
+		step := ReportStep{
+			Deduction: decimal.Zero,
+		}
+
+		defer func() {
+			r.steps[ref] = step
```

Comment:
This ensure all step data is applied for the `Deduct` call no matter the failure cases.